present/pyPresent.py

```diff
@@ -193,7 +203,11 @@ def string2number(i):
     Input: string (big-endian)
     Output: long or integer
     """
-    return int(i.encode('hex'), 16)
+    from codecs import getencoder
+    encoder_hex = getencoder('hex')
+    if type(i) == bytes:
+        return int(encoder_hex(i)[0], 16)
+    return int(encoder_hex(bytes(i.encode("UTF-8")))[0], 16)
```

Comment:
I'm not really sure how I feel about this change. Looks like it's way more complex now